onnx/backend/test/case/node/__init__.py

```diff
@@ -187,8 +208,27 @@ def expect(node: onnx.NodeProto,
         name=name,
         inputs=inputs_vi,
         outputs=outputs_vi)
-    kwargs[str('producer_name')] = 'backend-test'
-    model = onnx.helper.make_model(graph, **kwargs)
+    kwargs[str("producer_name")] = "backend-test"
+
+    if "opset_imports" not in kwargs:
+        # To make sure the model will be produced with the same opset_version after opset changes
+        # By default, it uses since_version as opset_version for produced models
+        if _TargetOpsetVersion is None:
+            produce_opset_version = onnx.defs.get_schema(node.op_type, node.domain).since_version
+        # If given targeted opset version
+        # it will generate test data for the latest opset vesion that supports before targeted opset version
+        else:
+            produce_opset_version = onnx.defs.get_schema(node.op_type, int(_TargetOpsetVersion), node.domain).since_version
+        kwargs[str("opset_imports")] = [onnx.helper.make_operatorsetid(node.domain, produce_opset_version)]
+    else:
+        # Also, for the same reason above, convert given opset.version (from opset_imports) to max_inclusive_version
```

Comment:
Do we really need to do this? If the user explicitly specifies opset version in the test-case, why not just use that? Automatically adjusting it can be problematic in some edge cases (eg., what if the user really wants a specific opset-version? Even though the test-case is for a single node, that node could be a control-flow node with a subgraph of operators which may require specific opset version, for example). On the flip side, does adjusting it give us any benefit or advantage?

Reply:
Thank you @gramalingam for catching it. I agree with your point -- If specifying opset version here, we should simply use it instead of using max_inclusive_version. It provides more flexibility for users if they really want specific opset version, especially for some control-flow cases. I will update here soon to use the opset version directly.